src/base/sampler.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <memory>
+
+#include "vector.h"
+
+namespace gm {
+
+/// Interface class for creating different types of samplers.
+/// These samplers should output random samples based on a
+/// uniform distribution in the range of [0, 1)^n where n is
+/// the number of dimensions. The samplers should create unique
+/// sequences based on the pixel that is being sampled.
+
+class Sampler {
+ public:
+  /// The Sampler is initialized with the number of samples per pixels.
+  /// Because we anticipate having to generate a large amount of samples
+  /// per pixel we store the spp inside a 64-bit integer
+  /// TODO This should be user-defined in the future.
+  Sampler(int64_t samplesPerPixel)
```

Comment:
Do we really want 64bits-worth of spp? Don't you think the 2  Billion spp we get with 32bits are enough?

Reply:
Yeah, good point. I'll change that out